lib/puppet/parser/resource.rb

```diff
@@ -199,6 +199,51 @@ def raw_tagged?(tag_array)
     super || ((scope_resource = scope.resource) && !scope_resource.equal?(self) && scope_resource.raw_tagged?(tag_array))
   end
 
+  # Fill resource params from a capability
```

Comment:
picky: uses imperative form "Fill", "Fills" is preferred. Also, it looks like it fills parameters from all entities that it is consuming.